src/fairseq2/models/jepa/factory.py

```diff
@@ -265,60 +291,108 @@ def build_encoder(self) -> TransformerEncoder:
             dtype=self._dtype,
         )
 
-    def build_encoder_layer(self) -> TransformerEncoderLayer:
+    def build_encoder_layer(self, layer_idx: int) -> TransformerEncoderLayer:
         config = self._config
 
-        self_attn = self.build_attention()
+        self_attn = self.build_attention(layer_idx)
 
-        ffn = self.build_ffn()
+        ffn = self.build_ffn(layer_idx)
+
+        drop_path = DropPathResidualConnect(drop_p=config.droppath_p)
 
         return StandardTransformerEncoderLayer(
             self_attn,
             ffn,
             dropout_p=config.dropout_p,
             norm_order=TransformerNormOrder.PRE,
             layer_norm_factory=self.build_layer_norm,
+            self_attn_residual=drop_path,
+            ffn_residual=drop_path,
             device=self._device,
             dtype=self._dtype,
         )
 
-    def build_attention(self) -> MultiheadAttention:
+    def build_attention(self, layer_idx: int) -> MultiheadAttention:
         config = self._config
 
         sdpa = create_default_sdpa(attn_dropout_p=config.attn_dropout_p)
 
+        output_proj = self.build_mha_output_projection(layer_idx)
+
         return StandardMultiheadAttention(
             config.model_dim,
             config.num_encoder_attn_heads,
             sdpa=sdpa,
             bias=config.qkv_bias,
-            output_proj_bias=True,
+            output_proj=output_proj,
             device=self._device,
             dtype=self._dtype,
         )
 
-    def build_ffn(self) -> FeedForwardNetwork:
+    def build_mha_output_projection(self, layer_idx: int) -> Linear:
         config = self._config
 
+        init_std = config.init_std
+
+        def init_projection(proj: Linear) -> None:
+            init_module(proj, std=init_std)
+
+            with torch.no_grad():
+                proj.weight.div_(math.sqrt(2.0 * layer_idx))
```

Comment:
Is this accurate? In the reference implementation, I see that the scaling is done with `layer_idx + 1` instead of `layer_idx`. https://github.com/facebookresearch/jepa/blob/main/src/models/vision_transformer.py#L150

Reply:
good catch , yes this was the mistake, thanks @cbalioglu